### DISCUSSION TITLE
How to define macros from meson command line invocation ?

### DISCUSSION THREAD

**Discussion by frak0d** [mesonbuild/meson#10608] [Q&A] [answered]
How to define macros from meson command line invocation ? Is there something like `meson --define:MY_MACRO=45` ?

**Comment by eli-schwartz** [comment]
If you're trying to pass macros to the C/C++ compiler, the project itself should probably be handling that via `add_project_arguments()`.

You can add it yourself via `c_args` or `cpp_args` though, which take an array of user (not project) -specified compiler flags.

**Reply by frak0d** [reply]
I need to do it from command line. Not from meson.build

**Reply by eli-schwartz** [reply]
And I told you that this is possible via the command line's `c_args` option.

**Reply by WarrenWeckesser** [reply]
@eli-schwartz can you given an example of a command line invocation that uses `cpp_args` to define a macro?  Not all of us have the meson command line API memorized!

**Reply by eli-schwartz** [reply, marked as answer]
```
CXXFLAGS="$CXXFLAGS -DMY_MACRO=45" meson setup builddir

meson setup builddir -Dcpp_args="$CXXFLAGS -DMY_MACRO=45"
```

**Reply by WarrenWeckesser** [reply]
Thanks!
